crmsh/utils.py

```diff
@@ -2292,4 +2149,255 @@ def interface_choice():
+    """
+    Class to get some properties of IP address
+    """
+
+    def __init__(self, addr):
+        """
+        Init function
+        """
+        self.addr = addr
+
+    @property
+    def ip_address(self):
+        """
+        Create ipaddress instance
+        """
+        return ipaddress.ip_address(self.addr)
+
+    @property
+    def version(self):
+        """
+        Get IP address version
+        """
+        return self.ip_address.version
+
+    @classmethod
+    def is_mcast(cls, addr):
+        """
+        Check whether the address is multicast address
+        """
+        cls_inst = cls(addr)
+        return cls_inst.ip_address.is_multicast
+
+    @classmethod
+    def is_ipv6(cls, addr):
+        """
+        Check whether the address is IPV6 address
+        """
+        return cls(addr).version == 6
+
+    @classmethod
+    def is_valid_IP(cls, addr):
+        """
+        Check whether the address is valid IP address
+        """
+        cls_inst = cls(addr)
+        try:
+            cls_inst.ip_address
+        except ValueError:
+            return False
+        else:
+            return True
+
+    @property
+    def is_loopback(self):
+        """
+        Check whether the address is loopback address
+        """
+        return self.ip_address.is_loopback
+
+    @property
+    def is_link_local(self):
+        """
+        Check whether the address is link-local address
+        """
+        return self.ip_address.is_link_local
+
+
+class Interface(IP):
+    """
+    Class to get information from one interface
+    """
+
+    def __init__(self, ip_with_mask):
+        """
+        Init function
+        """
+        self.ip, self.mask = ip_with_mask.split('/')
+        super(__class__, self).__init__(self.ip)
+
+    @property
+    def ip_with_mask(self):
+        """
+        Get ip with netmask
+        """
+        return '{}/{}'.format(self.ip, self.mask)
+
+    @property
+    def ip_interface(self):
+        """
+        Create ip_interface instance
+        """
+        return ipaddress.ip_interface(self.ip_with_mask)
+
+    @property
+    def network(self):
+        """
+        Get network address
+        """
+        return str(self.ip_interface.network.network_address)
+
+    def ip_in_network(self, addr):
+        """
+        Check whether the addr in the network
+        """
+        return IP(addr).ip_address in self.ip_interface.network
+
+
+class InterfacesInfo(object):
+    """
+    Class to collect interfaces information on local node
+    """
+
+    def __init__(self, ipv6=False, second_heartbeat=False, custom_nic_list=[]):
+        """
+        Init function
+
+        On init process,
+        "ipv6" is provided by -I option
+        "second_heartbeat" is provided by -M option
+        "custom_nic_list" is provided by -i option
+        """
+        self.ip_version = 6 if ipv6 else 4
+        self.second_heartbeat = second_heartbeat
+        self._default_nic_list = custom_nic_list
```

Comment:
I don't understand much why you set the `custom_nic_list` to `default_nic_list`. Why don't you just keep `self._nic_list` or similar. What's the meaning of `default` and `custom` here?